src/pipecat/processors/transcript_processor.py

```diff
@@ -0,0 +1,235 @@
+
+                if text_parts:
+                    result.append(
+                        TranscriptionMessage(role="assistant", content=" ".join(text_parts))
+                    )
+
+        return result
+
+    def _find_new_messages(self, current: List[TranscriptionMessage]) -> List[TranscriptionMessage]:
+        """Find unprocessed messages from current list.
+
+        Args:
+            current: List of current messages
+
+        Returns:
+            List[TranscriptionMessage]: New messages not yet processed
+        """
+        if not self._processed_messages:
+            return current
+
+        processed_len = len(self._processed_messages)
+        if len(current) <= processed_len:
+            return []
+
+        return current[processed_len:]
+
+    async def process_frame(self, frame: Frame, direction: FrameDirection):
+        """Process frames into assistant conversation messages.
+
+        Args:
+            frame: Input frame to process
+            direction: Frame processing direction
+        """
+        await super().process_frame(frame, direction)
+
+        if isinstance(frame, OpenAILLMContextFrame):
+            standard_messages = []
+            for msg in frame.context.messages:
+                converted = frame.context.to_standard_messages(msg)
+                standard_messages.extend(converted)
+
+            current_messages = self._extract_messages(standard_messages)
+            new_messages = self._find_new_messages(current_messages)
+            self._pending_assistant_messages.extend(new_messages)
+
+        elif isinstance(frame, OpenAILLMContextAssistantTimestampFrame):
+            if self._pending_assistant_messages:
+                for msg in self._pending_assistant_messages:
+                    msg.timestamp = frame.timestamp
+                await self._emit_update(self._pending_assistant_messages)
+                self._pending_assistant_messages = []
+
+        await self.push_frame(frame, direction)
+
+
+class TranscriptProcessor:
+    """Factory for creating and managing transcript processors.
+
+    Provides unified access to user and assistant transcript processors
+    with shared event handling.
+
+    Example:
+        ```python
+        transcript = TranscriptProcessor()
+
+        pipeline = Pipeline(
+            [
+                transport.input(),
+                stt,
+                transcript.user(),              # User transcripts
+                context_aggregator.user(),
+                llm,
+                tts,
+                transport.output(),
+                context_aggregator.assistant(),
+                transcript.assistant(),         # Assistant transcripts
+            ]
+        )
+
+        @transcript.event_handler("on_transcript_update")
+        async def handle_update(processor, frame):
+            print(f"New messages: {frame.messages}")
+        ```
+    """
+
+    def __init__(self, **kwargs):
```

Comment:
We probably don't want to pass `kwargs` here. Both processors might have different arguments and we would not be able to pass them like this.